tools/server/server.cpp

```diff
@@ -3662,6 +3662,7 @@ struct server_context {
                 completion_token_output result;
                 result.tok          = id;
                 result.text_to_send = common_token_to_piece(ctx, result.tok, accept_special_token(slot, result.tok));
+                fprintf(stderr, "[DETOKENIZE] Token ID: %d -> Text: '%s' (length: %zu)\n", result.tok, result.text_to_send.c_str(), result.text_to_send.length());
```

Comment:
I love adding useful logs like this! This project has its own set of logging macros to use. In the core of the project, you've got `LLAMA_LOG_*` (defined [here](https://github.com/ggml-org/llama.cpp/blob/master/src/llama-impl.h#L26)). In the `server` tool, there are three sets defined [here](https://github.com/ggml-org/llama.cpp/blob/master/tools/server/utils.hpp#L34): `SVR_*` for server-level logs, `SLT_*` for slot-level logs, and `QUE_*` for queue-level logs.